### PR TITLE
test: check reporting user sends an email

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref: https://github.com/freeCodeCamp/freeCodeCamp/issues/50635

### CODE REVIEW

File: e2e/utils/mailhog.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+export const getSubject = (email: {
+  Content: { Headers: { Subject: string[] } };
+}) => {
+  return email.Content.Headers.Subject[0];
+};
+
+export const deleteAllEmails = async () => {
+  await fetch(`http://${host}:8025/api/v1/messages`, {
```
Comment (naomi-lgbt) [inline e2e/utils/mailhog.ts:27]: Why are we using two different API versions?

Comment (ojeytonwilliams) [reply e2e/utils/mailhog.ts:27]: Weirdly delete doesn't exist in V2. At least not according to their spec, I didn't actually check.

Comment (ojeytonwilliams) [reply e2e/utils/mailhog.ts:27]: Okay, I've checked. DELETE /api/v1/messages works, but DELETE /api/v2/messages is a 404 :shrug:

Comment (naomi-lgbt) [reply e2e/utils/mailhog.ts:27]: 😭

**Review by naomi-lgbt — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**